sycl/include/syclcompat/dims.hpp

```diff
@@ -38,6 +38,8 @@ class dim3 {
 
   dim3(const sycl::range<1> &r) : x(r[0]), y(1), z(1) {}
 
+  constexpr dim3() : x(0), y(0), z(0) {}
```

Comment:
Do you need to set this to zero. The value should not be used and I assume all you need here is to be able to instantiate this class and set the dim later on. Can you set it to   `dim3() = default;`